modules/azure/compute.go

```diff
@@ -0,0 +1,102 @@
+package azure
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute"
+	"github.com/stretchr/testify/require"
+)
+
+// getVirtualMachineClient is a helper function that will setup an Azure Virtual Machine client on your behalf
+func getVirtualMachineClient(subscriptionID string) (compute.VirtualMachinesClient, error) {
+	// Validate Azure subscription ID
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return compute.NewVirtualMachinesClient(""), err
```

Comment:
Is `compute.VirtualMachinesClient` an interface or pointer? Can you return `nil` here instead of instantiating one?

Reply:
I can refactor my approach here to return `nil`.  Will take care of it in next commit.